lib/autokey/service.py

```diff
@@ -489,28 +497,37 @@ def execute(self, script: model.Script, buffer=''):
         if script.path is not None:
             # Overwrite __file__ to contain the path to the user script instead of the path to this service.py file.
             scope["__file__"] = script.path
-        try:
-            exec(script.code, scope)
-        except Exception as e:
-            logger.exception("Script error")
-            self.error = "Script name: '{}'\n{}".format(script.description, traceback.format_exc())
-            self.app.notify_error("The script '{}' encountered an error".format(script.description))
+        self._execute(scope, script)
 
         self.mediator.send_string(trigger_character)
 
     @threaded
-    def execute_path(self, path: pathlib.Path, buffer=''):
+    def execute_path(self, path: pathlib.Path):
         logger.debug("Script runner executing: {}".format(path))
         scope = self.scope.copy()
-        # scope["store"] = script.store
         # Overwrite __file__ to contain the path to the user script instead of the path to this service.py file.
         scope["__file__"] = path.resolve()
+        self._execute(scope, path)
+
+    def _record_error(self, error: model.ScriptErrorRecord):
+        self.error_records.append(error)
+        self.app.notify_error(error)
+
+    def _execute(self, scope, script: typing.Union[model.Script, pathlib.Path]):
+        script_code = script.read_text() if isinstance(script, pathlib.Path) else script.code
+        start_time = datetime.datetime.now().time()
+        # noinspection PyBroadException
         try:
-            exec(path.read_text(), scope)
+            exec(script_code, scope)
         except Exception as e:
+            error_time = datetime.datetime.now().time()
```

Comment:
The exception block could possibly be its own function?

Reply:
I can move everything into `_record_error()`, which currently doesn’t do that much:
https://github.com/autokey/autokey/blob/5fcbe73d35f4c41718c57dfda37a79a8441e29cb/lib/autokey/service.py#L512-L514

Reply:
Good